src/legacy/ui/public/index_patterns/_index_pattern.js

```diff
@@ -119,24 +117,35 @@ export function IndexPatternProvider(Private, config, Promise, confirmModalPromi
     }
 
     if (indexPattern.isUnsupportedTimePattern()) {
-      if (!isUserAwareOfUnsupportedTimePattern(indexPattern)) {
-        const warningTitle = i18n.translate('common.ui.indexPattern.warningTitle', {
-          defaultMessage: 'Support for time intervals was removed',
-        });
+      const warningTitle = i18n.translate('common.ui.indexPattern.warningTitle', {
+        defaultMessage: 'Support for time interval index patterns removed',
+      });
 
-        const warningText = i18n.translate('common.ui.indexPattern.warningText', {
-          defaultMessage: 'For more information, view the ["{title}" index pattern in management]({link})',
-          values: {
-            title: indexPattern.title,
-            link: kbnUrl.getRouteHref(indexPattern, 'edit'),
-          },
-        });
+      const warningText = i18n.translate('common.ui.indexPattern.warningText', {
+        defaultMessage: 'Instead of querying indices only matching the pattern {title}, you are now querying all indices ' +
+          'matching {index}. This index pattern should be migrated to a wildcard-based index pattern. To do so, edit the ' +
+          'index pattern in Management.',
+        values: {
+          title: indexPattern.title,
+          index: indexPattern.getIndex()
+        }
+      });
 
-        toastNotifications.addWarning({
-          title: warningTitle,
-          text: warningText,
-        });
-      }
+      toastNotifications.addWarning({
+        title: warningTitle,
+        text: (
+          <div>
+            <p>{warningText}</p>
+            <EuiFlexGroup justifyContent="flexEnd" gutterSize="s">
+              <EuiFlexItem grow={false}>
+                <EuiButton size="s" href={kbnUrl.getRouteHref(indexPattern, 'edit')}>
+                  Edit index pattern
+                </EuiButton>
```

Comment:
UX nit: Maybe this is something to add later, but I wonder if we should be integrating with feature controls here. That way we could hide the edit button if we know a user doesn't have access to management, rather than allowing them to click a button that will lead straight to an error page.